code/game/objects/items/twohanded.dm

```diff
@@ -962,3 +963,169 @@
+	icon = 'icons/obj/weapons/misc.dmi'
+	icon_state = "vxtvul_hammer0-0"
+	lefthand_file = 'icons/mob/inhands/weapons/hammers_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/hammers_righthand.dmi'
+	name = "Vxtvul Hammer"
+	desc = "A relict sledgehammer with charge packs wired to two blast pads on its head. \
+			While wielded in two hands, the user can charge a massive blow that will shatter construction and hurl bodies."
+	force = 4 //It's heavy as hell
+	force_wielded = 24 
+	armour_penetration = 50 //Designed for shattering walls in a single blow, I don't think it cares much about armor
+	throwforce = 18
+	attack_verb = list("attacked", "hit", "struck", "bludgeoned", "bashed", "smashed")
+	sharpness = SHARP_NONE //Blunt, breaks bones
+	wound_bonus = -10
+	bare_wound_bonus = 15
+	max_integrity = 200
+	resistance_flags = ACID_PROOF | FIRE_PROOF
+	w_class = WEIGHT_CLASS_HUGE
+	slot_flags = ITEM_SLOT_BACK
+	actions_types = list(/datum/action/item_action/charge_hammer)
+	var/datum/effect_system/spark_spread/spark_system //It's a surprise tool that'll help us later
+	var/is_charging = 0
+	var/is_supercharged = 0
+
+/obj/item/twohanded/vxtvulhammer/Initialize() //For the sparks when you begin to charge it
+	. = ..()
+	spark_system = new
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+
+/obj/item/twohanded/vxtvulhammer/Destroy() //Even though the hammer won't probably be destroyed, Ever™
+	qdel(spark_system)
+	spark_system = null
+	return ..()
+
+/obj/item/twohanded/vxtvulhammer/update_icon()
+	icon_state = "vxtvul_hammer[wielded]-[is_supercharged]"
+
+/obj/item/twohanded/vxtvulhammer/examine(mob/living/carbon/user)
+	. = ..()
+	if(is_supercharged == 1)
+		. += "<b>Electric sparks</b> are bursting from the blast pads!"
+
+/obj/item/twohanded/vxtvulhammer/unwield(mob/living/carbon/user)
+	..()
+	if(is_supercharged == 1) //So you can't one-hand the charged hit
+		to_chat(user, span_notice("Your hammer loses its power as you adjust your grip."))
+		user.visible_message(span_warning("The sparks from [user]'s hammer suddenly stop!"))
+		supercharge(user)
+	if(is_charging == 1) //So you can't one-hand while charging
+		to_chat(user, span_notice("You flip the switch off as you adjust your grip."))
+		user.visible_message(span_warning("[user] flicks the hammer off!"))
+		is_charging = 0
+
+/obj/item/twohanded/vxtvulhammer/attack(atom/target, mob/living/carbon/user) //This doesn't consider objects, only people
+	if (is_charging == 1) //So you can't attack while charging
+		to_chat(user, span_notice("You flip the switch off before your attack."))
+		user.visible_message(span_warning("[user] flicks the hammer off and raises it!"))
+		is_charging = 0
+	..()
+
+/obj/item/twohanded/vxtvulhammer/AltClick(mob/living/carbon/user)
+	charge_hammer(user)
+
+/obj/item/twohanded/vxtvulhammer/proc/supercharge(mob/living/carbon/user) //Proc to handle when it's charged for light + sprite
+	if(is_supercharged == 0)
+		is_supercharged = 1
```

Comment:
my lord baby what is you doing

Reply:
Putting out a draft of work to optimize this shitcode rather than saying "merge this".

If you have anything actually constructive to say or suggest I would love that instead of just "lol".